src/libraries/System.ComponentModel.Composition/src/Microsoft/Internal/Collections/WeakReferenceCollection.cs

```diff
@@ -18,7 +25,10 @@ public void Add(T item)
                 CleanupDeadReferences();
             }
 
-            _items.Add(new WeakReference(item));
+            using (_lock.LockComposition())
```

Comment:
It looks better to use the `narrow lock` as it is not changing a composition
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Hosting/CompositionLock.cs#L11-L12
```suggestion
            using (_lock.LockStateForWrite())
```